benches/bloom_bench.rs

```diff
@@ -0,0 +1,64 @@
+use bytes::{BufMut, BytesMut};
+use criterion::{black_box, criterion_group, criterion_main, Criterion};
+use slatedb::filter::{filter_hash, BloomFilterBuilder};
+fn bloom_filter_insert_benchmark(c: &mut Criterion) {
+    let mut builder = BloomFilterBuilder::new(10); // Set bits_per_key
+    let key_sz = std::mem::size_of::<u32>();
+
+    c.bench_function("bloom_filter insert 10000000 keys", |b| {
+        b.iter(|| {
+            for i in 0..10000000 {
+                let mut bytes = BytesMut::with_capacity(key_sz);
+                bytes.put_u32(i);
+                builder.add_key(black_box(bytes.freeze().as_ref())); // Use `black_box` to prevent compiler optimizations
+            }
+        })
+    });
+}
+fn bloom_filter_build_benchmark(c: &mut Criterion) {
+    let mut builder = BloomFilterBuilder::new(10);
+    let key_sz = std::mem::size_of::<u32>();
+
+    // Add keys to builder before measuring build time
+    for i in 0..100000 {
+        let mut bytes = BytesMut::with_capacity(key_sz);
+        bytes.put_u32(i);
+        builder.add_key(bytes.freeze().as_ref());
+    }
+
+    c.bench_function("bloom_filter build", |b| {
+        b.iter(|| {
+            let _filter = black_box(builder.build());
+        })
+    });
+}
```

Comment:
It would be good to have a benchmark that tests the performance of building the filter for an entire SST - e.g. combining the above two benchmarks into one that both inserts keys and builds the filter. That (alongside querying the filter) is really what we care about optimizing for filters since that's what the calling code does. The specific benchmarks for insert and build are nice if we see that the combined benchmark is spending a lot of time in either place and we want to see if an optimization targeting insert or build worked, but I think what we'd want to measure to produce a reference number (e.g. for regression tests) is building a whole filter.